src/state.tsx

```diff
@@ -556,6 +557,14 @@ class GlobalState implements State {
     }
   }
 
+  renameLoadout(ix: number, name: string) {
+    const loadout = this.loadouts[ix];
+
+    if (loadout) {
+      loadout.name = name;
```

Comment:
I think we should strip whitespace here, and don't allow the rename to happen if `name` is an empty string. Ideally, we also need to give feedback to the user about why we didn't do the rename.

Reply:
I stripped the whitespace and prevent the rename if the name isn't valid. I also added validation to the input so the user will see if the title is valid or not before they submit.